src/KafkaClient/Protocol/KafkaEncoder.cs

```diff
@@ -145,7 +145,7 @@ private static IResponse ProduceResponse(IRequestContext context, ArraySegment<b
                         var offset = reader.ReadInt64();
                         DateTimeOffset? timestamp = null;
 
-                        if (context.ApiVersion >= 2) {
+                        if (context.ApiVersion >= 3) {
```

Comment:
timestamp is part of version 2 (there is no 3 currently):

Produce Response (Version: 2) => [responses] throttle_time_ms 
  responses => topic [partition_responses] 
    topic => STRING
    partition_responses => partition error_code base_offset timestamp 
      partition => INT32
      error_code => INT16
      base_offset => INT64
      timestamp => INT64
  throttle_time_ms => INT32